neuralhydrology/training/__init__.py

```diff
@@ -30,6 +30,8 @@ def get_optimizer(model: torch.nn.Module, cfg: Config) -> torch.optim.Optimizer:
     """
     if cfg.optimizer.lower() == "adam":
         optimizer = torch.optim.Adam(model.parameters(), lr=cfg.learning_rate[0])
+    elif cfg.optimizer.lower() == "adamw":
+        optimizer = torch.optim.AdamW(model.parameters(), lr=cfg.learning_rate[0])
```

Comment:
Is the default weight_decay parameter good? Haven't used AdamW much but I wonder if we want to make this a configurable parameter or not. @gauchm WDYT?

Reply:
In my experience, the default parameter is good, and doesn't need to be configured like learning rate does. Theoretically weight decay is best for preventing overfitting, so if you think that is likely to be a problem, then I vote to make it configurable.

Reply:
No strong opinion either way, but for me it's fine to leave it as is for now. If we start on that, we might also want to make the betas configurable and then it already get quite a lot of new config args. It's easy to hack in for someone who really wants to try it.

Reply:
SGTM